### ISSUE TITLE
sql: Support "CREATE DATABASE foo WITH TEMPLATE bar" syntax

### ISSUE THREAD

**Issue by alex** [cockroachdb/cockroach#10151]
- What did you do?

Attempted to run: `CREATE DATABASE "test__1" WITH TEMPLATE "test_"`

- What did you expect to see?

A successful creation of a new database

- What did you see instead?

```
syntax error at or near "WITH"
CREATE DATABASE "test__1" WITH TEMPLATE "test_"
```

**Comment by knz** [comment]
Hi @alex thanks for reporting this issue. Could you tell us a bit more about the context in which you'd find this feature useful?

**Comment by alex** [comment]
This feature is used by Django's test runner.

**Comment by knz** [comment]
Thanks!

**Comment by knz** [comment]
cc @nvanbenschoten did you see this come up in other clients too?

**Comment by nvanbenschoten** [comment]
I have never seen this used elsewhere. If it's only ever used for testing, I don't think it needs to be prioritized very highly. That said, it would be nice to get Django's tests working against us, and I don't think this specific syntax would be very difficult to support.

**Comment by alex** [comment]
To be clear, this is used both in Django's own test suites, as well as the test suites for any projects using Django.

**Comment by mjibson** [comment]
Yes, I think our upcoming work on backup/restore would handle this case in a graceful way. However I would guess we are some months off from worrying about this specific use case because other work is a higher priority. If we have many django projects blocked on this feature we could reprioritize.

**Comment by bdarnell** [comment]
I don't think this is the only thing blocking django support (last time I looked into this, the wall I ran into #5950)

This is a case where it's probably easier to use a custom cockroachdb backend for the django ORM than to implement all the corners of sql the same way postgres does. `CREATE DATABASE WITH TEMPLATE` is a postgres extension; django doesn't use it for other databases. It just needs some way to clone a database (for mysql, it [shells out to mysqldump](https://github.com/django/django/blob/master/django/db/backends/mysql/creation.py)). With a subclass of the postgresl backend we could make it walk over `SHOW TABLE` and use `CREATE TABLE $1 AS SELECT * FROM $2`

**Comment by tbo** [comment]
Shouldn't the documentation reflect that this isn't supported? It is listed in the [sql grammar documentation](https://www.cockroachlabs.com/docs/stable/sql-grammar.html#opt_template_clause) so I assumed it to be working.

**Comment by ajwerner** [comment]
> Shouldn't the documentation reflect that this isn't supported? It is listed in the [sql grammar documentation](https://www.cockroachlabs.com/docs/stable/sql-grammar.html#opt_template_clause) so I assumed it to be working.

@nickvigilante any interest in documenting this?

**Comment by nickvigilante** [comment]
@ajwerner Yep! I just created [this JIRA ticket](https://cockroachlabs.atlassian.net/browse/DOC-2370).

**Comment by pschichtel** [comment]
This seems to do the trick:
```sql
BACKUP DATABASE source_db INTO 'userfile://defaultdb.public.userfiles_root/copy';
RESTORE DATABASE source_db FROM LATEST IN 'userfile://defaultdb.public.userfiles_root/copy' WITH new_db_name = 'target';
```

Or am I missing something?

**Comment by Bessonov** [comment]
@pschichtel Thank you for providing the workaround!

> Or am I missing something?

Yes, there is a mistake in the documentation, and there's a lack of compatibility with PostgreSQL.
